packages/edit-site/src/components/save-hub/index.js

```diff
@@ -1,186 +1,42 @@
+		const _isSaving = dirtyEntityRecords.some( ( record ) =>
+			isSavingEntityRecord( record.kind, record.name, record.key )
 		);
-
-	const {
-		editEntityRecord,
-		saveEditedEntityRecord,
-		__experimentalSaveSpecifiedEntityEdits: saveSpecifiedEntityEdits,
-	} = useDispatch( coreStore );
-
-	const disabled = isSaving || ( ! isDirty && ! isPreviewingTheme() );
-
-	// if we have only one unsaved change and it matches current context, we can show a more specific label
-	let label = dirtyCurrentEntity
-		? __( 'Save' )
-		: sprintf(
-				// translators: %d: number of unsaved changes (number).
-				_n(
-					'Review %d change…',
-					'Review %d changes…',
-					countUnsavedChanges
-				),
-				countUnsavedChanges
-		  );
-
-	if ( isSaving ) {
-		label = __( 'Saving' );
-	}
-
-	const { homeUrl } = useSelect( ( select ) => {
-		const {
-			getUnstableBase, // Site index.
-		} = select( coreStore );
 		return {
-			homeUrl: getUnstableBase()?.home,
+			isSaving: _isSaving,
+			isDisabled:
+				_isSaving ||
+				( ! dirtyEntityRecords.length && ! isPreviewingTheme() ),
 		};
 	}, [] );
-
-	const saveCurrentEntity = async () => {
-		if ( ! dirtyCurrentEntity ) return;
-
-		removeNotice( saveNoticeId );
-		const { kind, name, key, property } = dirtyCurrentEntity;
-
-		try {
-			if ( 'root' === dirtyCurrentEntity.kind && 'site' === name ) {
-				await saveSpecifiedEntityEdits( 'root', 'site', undefined, [
-					property,
-				] );
-			} else {
-				if (
-					PUBLISH_ON_SAVE_ENTITIES.some(
-						( typeToPublish ) =>
-							typeToPublish.kind === kind &&
-							typeToPublish.name === name
-					)
-				) {
-					editEntityRecord( kind, name, key, { status: 'publish' } );
-				}
-
-				await saveEditedEntityRecord( kind, name, key );
-			}
-
-			__unstableMarkLastChangeAsPersistent();
-
-			createSuccessNotice( __( 'Site updated.' ), {
-				type: 'snackbar',
-				actions: [
-					{
-						label: __( 'View site' ),
-						url: homeUrl,
-					},
-				],
-				id: saveNoticeId,
-			} );
-		} catch ( error ) {
-			createErrorNotice( `${ __( 'Saving failed.' ) } ${ error }` );
-		}
-	};
-
 	return (
 		<HStack className="edit-site-save-hub" alignment="right" spacing={ 4 }>
-			{ dirtyCurrentEntity ? (
-				<Button
-					variant="primary"
-					onClick={ saveCurrentEntity }
-					isBusy={ isSaving }
-					disabled={ isSaving }
-					aria-disabled={ isSaving }
-					className="edit-site-save-hub__button"
-					__next40pxDefaultSize
-				>
-					{ label }
-				</Button>
-			) : (
-				<SaveButton
-					className="edit-site-save-hub__button"
-					variant={ disabled ? null : 'primary' }
-					showTooltip={ false }
-					icon={ disabled && ! isSaving ? check : null }
-					defaultLabel={ label }
-					__next40pxDefaultSize
-				/>
-			) }
+			<SaveButton
+				className="edit-site-save-hub__button"
+				variant={ isDisabled ? null : 'primary' }
+				showTooltip={ false }
+				icon={ isDisabled && ! isSaving ? check : null }
+				showReviewMessage
```

Comment:
This is the prop I mentioned in the PR's description to show a different message if there are more entities to update.

The `defaultLabel` which I removed, was introduced and used for this specific case only.